sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/implementation/ChangeFeedStartFromETagAndFeedRangeImpl.java

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.implementation.changefeed.implementation;
+
+import com.azure.cosmos.implementation.Constants;
+import com.azure.cosmos.implementation.HttpConstants;
+import com.azure.cosmos.implementation.RxDocumentServiceRequest;
+import com.azure.cosmos.implementation.feedranges.FeedRangeInternal;
+
+import static com.azure.cosmos.BridgeInternal.setProperty;
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkNotNull;
+
+class ChangeFeedStartFromETagAndFeedRangeImpl extends ChangeFeedStartFromInternal {
```

Comment:
This class corresponds to starting from either "null" or a given continuation state (aka ChangeFeedState), right?  It would be useful to add a short description to all these ChangeFeedStart* classes which captures the specific cases they will be used.

Reply:
This one is used to continue changefeed query when you just have server continuation and it is for a single range (like with the Lease document in ChangeFeedProcessor) Added a comment along those lines (next iteration) - the remaining StartFrom settings have self-explanatory names.